homeassistant/components/qnap/sensor.py

```diff
@@ -284,62 +271,26 @@ def round_nicely(number):
     return round(number)
 
 
-class QNAPStatsAPI:
-    """Class to interface with the API."""
-
-    def __init__(self, config):
-        """Initialize the API wrapper."""
-
-        protocol = "https" if config[CONF_SSL] else "http"
-        self._api = QNAPStats(
-            f"{protocol}://{config.get(CONF_HOST)}",
-            config.get(CONF_PORT),
-            config.get(CONF_USERNAME),
-            config.get(CONF_PASSWORD),
-            verify_ssl=config.get(CONF_VERIFY_SSL),
-            timeout=config.get(CONF_TIMEOUT),
-        )
-
-        self.data = {}
-
-    @Throttle(MIN_TIME_BETWEEN_UPDATES)
-    def update(self):
-        """Update API information and store locally."""
-        try:
-            self.data["system_stats"] = self._api.get_system_stats()
-            self.data["system_health"] = self._api.get_system_health()
-            self.data["smart_drive_health"] = self._api.get_smart_disk_health()
-            self.data["volumes"] = self._api.get_volumes()
-            self.data["bandwidth"] = self._api.get_bandwidth()
-        except Exception:  # pylint: disable=broad-except
-            _LOGGER.exception("Failed to fetch QNAP stats from the NAS")
-
-
-class QNAPSensor(SensorEntity):
+class QNAPSensor(CoordinatorEntity, SensorEntity):
     """Base class for a QNAP sensor."""
 
     def __init__(
-        self, api, description: SensorEntityDescription, monitor_device=None
+        self, coordinator, description: SensorEntityDescription, monitor_device=None
     ) -> None:
         """Initialize the sensor."""
+        self.coordinator = coordinator
         self.entity_description = description
         self.monitor_device = monitor_device
-        self._api = api
+        self.device_name = self.coordinator.data["system_stats"]["system"]["name"]
 
     @property
     def name(self):
         """Return the name of the sensor, if any."""
-        server_name = self._api.data["system_stats"]["system"]["name"]
-
         if self.monitor_device is not None:
             return (
-                f"{server_name} {self.entity_description.name} ({self.monitor_device})"
+                f"{self.device_name} {self.monitor_device} - {self.entity_description.name}"
```

Comment:
Why did you swap description and monitor device?
```suggestion
                f"{self.device_name} {self.entity_description.name} ({self.monitor_device})"
```